src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,207 @@
+package duke.parser;
+
+import duke.task.*;
+
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Scanner;
+
+/**
+ *  This Class is mainly dealing with user input and logic process
+ */
+public class Parser {
+    private String input;
+    private String[] inputArray;
+
+    /**
+     * Constructs a Parser based on the user input and register user's input and sliced input
+     *
+     * @param input input from user
+     */
+    public Parser(String input) {
+        this.input = input;
+        inputArray= this.input.split(" ");
+    }
+
+    /**
+     *  Return input that has been pre-processed become sliced Array
+     *
+     * @return inputArray
+     */
+    public String[] parseInput() {
+
+        return inputArray;
+    }
+
+
+    /**
+     *  Return command after parsing the input
```

Comment:
A lot of the comments have an extra whitespace character, consider removing it for standardisation.
```suggestion
     * Return command after parsing the input
```